### PR TITLE
Test for cpu inference

### PR DESCRIPTION
**Pull request by lukeyeager**
Add a test to detect this bug: https://github.com/NVIDIA/caffe/issues/196 (was https://github.com/NVIDIA/DIGITS/issues/907)

Verifies this fix: https://github.com/NVIDIA/caffe/pull/200

### CODE REVIEW

**Comment by lukeyeager** [comment]
@gheinrich can you test this with Torch? It isn't working for me but I don't understand the errors.

File: digits/task.py [hunk 1]
```diff
@@ -203,7 +203,9 @@ def run(self, resources):
         #https://docs.python.org/2/library/subprocess.html#converting-argument-sequence
         if platform.system() == 'Windows':
             args = ' '.join(args)
-        self.logger.info('Task subprocess args: {}'.format(args))
+            self.logger.info('Task subprocess args: "{}"'.format(args))
+        else:
+            self.logger.info('Task subprocess args: "%s"' % ' '.join(args))
```
Comment (gheinrich) [inline digits/task.py]: Thanks I always make this change locally to be able to copy/paste command lines into a command prompt! Otherwise you get the list notation (with brackets, commas et al.).

**Comment by gheinrich** [comment]
Hi Luke, I added a couple of commits on top of yours on https://github.com/gheinrich/DIGITS/tree/dev/torch-cpu-inference-test. This addresses the case where you train a model on GPU then perform inference on CPU **under some condition**. The condition being that you didn't use CuDNN layers during training as it is otherwise pretty difficult to convert the model back to non-CuDNN. I'd like to skip the test in this case (as I did for LeNet test cases). We can revisit this if we determine later on that we desperately need to do this.

**Comment by lukeyeager** [comment]
Added your changes, thanks!

And thanks for the info. I didn't realize we had those sorts of problems with Torch. I believe we have similar issues for Caffe with BatchNorm, too.
